heron/stmgr/tests/cpp/server/stmgr_unittest.cpp

```diff
@@ -1521,7 +1521,8 @@ TEST(StMgr, test_tmaster_restart_on_same_address) {
   // Note: Here we sleep longer compared to the previous test as we need
   // to tmasterClient could take upto 1 second (specified in test_heron_internals.yaml)
   // to retry connecting to tmaster.
-  sleep(3);
+  while (regular_stmgr->GetPhysicalPlan()->stmgrs(1).data_port() == common.stmgr_baseport_ + 1)
```

Comment:
30 seconds? It's hard to tell for sure, since I'm not familiar with the specific of what this test is doing. If you start with 30 seconds and we're still timing out we could increase it, or try to understand why.